transport/awslambda/handler.go

```diff
@@ -0,0 +1,120 @@
+package awslambda
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/go-kit/kit/log"
+)
+
+// Handler wraps an endpoint.
+type Handler struct {
+	e            endpoint.Endpoint
+	dec          DecodeRequestFunc
+	enc          EncodeResponseFunc
+	before       []HandlerRequestFunc
+	after        []HandlerResponseFunc
+	errorEncoder ErrorEncoder
+	finalizer    []HandlerFinalizerFunc
+	logger       log.Logger
+}
+
+// NewHandler constructs a new handler, which implements
+// the AWS lambda.Handler interface.
+func NewHandler(
+	e endpoint.Endpoint,
+	dec DecodeRequestFunc,
+	enc EncodeResponseFunc,
+	options ...HandlerOption,
+) *Handler {
+	h := &Handler{
+		e:            e,
+		dec:          dec,
+		enc:          enc,
+		logger:       log.NewNopLogger(),
+		errorEncoder: DefaultErrorEncoder,
+	}
+	for _, option := range options {
+		option(h)
+	}
+	return h
+}
+
+// HandlerOption sets an optional parameter for handlerh.
+type HandlerOption func(*Handler)
+
+// HandlerBefore functions are executed on the payload byte,
+// before the request is decoded.
+func HandlerBefore(before ...HandlerRequestFunc) HandlerOption {
+	return func(h *Handler) { h.before = append(h.before, before...) }
+}
+
+// HandlerAfter functions are only executed after invoking the endpoint
+// but prior to returning a response.
+func HandlerAfter(after ...HandlerResponseFunc) HandlerOption {
+	return func(h *Handler) { h.after = append(h.after, after...) }
+}
+
+// HandlerErrorLogger is used to log non-terminal errorh.
+// By default, no errors are logged.
+func HandlerErrorLogger(logger log.Logger) HandlerOption {
+	return func(h *Handler) { h.logger = logger }
+}
+
+// HandlerErrorEncoder is used to encode errorh.
+func HandlerErrorEncoder(ee ErrorEncoder) HandlerOption {
+	return func(h *Handler) { h.errorEncoder = ee }
+}
+
+// HandlerFinalizer sets finalizer which are called at the end of
+// request. By default no finalizer is registered.
+func HandlerFinalizer(f ...HandlerFinalizerFunc) HandlerOption {
+	return func(h *Handler) { h.finalizer = append(h.finalizer, f...) }
+}
+
+// DefaultErrorEncoder defines the default behavior of encoding an error response,
+// where it returns nil, and the error itself.
+func DefaultErrorEncoder(ctx context.Context, err error) ([]byte, error) {
```

Comment:
I think this code is now uncovered by the tests? Though I obviously don't expect it to have bugs 😉

Reply:
Add it in 1a313386a870a505534d47b3eb1a253eadb595b0 ☺️pretty important for future safe-guard from accidental breaking changes.